### PR TITLE
CORE:

### PR DESCRIPTION
**Pull request by manolama**
- Add the ability to store deltas in the anomaly prediction as a config
  flag.

EGADS:
- Compute the deltas.
- Add a cas check to the state updater

REDIS:
- Use setex to properly update the state.

### CODE REVIEW

File: core/src/main/java/net/opentsdb/query/router/TimeRouterFactory.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public void setupGraph(final QueryPipelineContext context,
     if (context.query().isDebugEnabled()) {
       StringBuilder buf = new StringBuilder();
       for (int i = 0; i < sources.size(); i++) {
-        if (i >= 0) {
+        if (i >= 0 && buf.length() > 0) {
```
Comment (gskchaitanya) [inline core/src/main/java/net/opentsdb/query/router/TimeRouterFactory.java]: `i > 0` should be enough here

File: implementation/egads/src/main/java/net/opentsdb/query/anomaly/egads/olympicscoring/OlympicScoringNode.java [hunk 2]
```diff
@@ -797,12 +821,35 @@ void updateState(final State new_state, final Exception e) {
...
+      try {
+        Thread.sleep(50);
+      } catch (InterruptedException e1) {
+        LOG.error("Unexpected interruption", e1);
+        return;
+      }
+      cache.setState(cache_key, state, 300_000); // TODO config
+      cas = cache.getState(cache_key);
+    }
+    if (cas.equals(state)) {
```
Comment (gskchaitanya) [inline implementation/egads/src/main/java/net/opentsdb/query/anomaly/egads/olympicscoring/OlympicScoringNode.java]: If another thread updates the cache just  after #845  and before #846, `cas` won't be equal to `state`. If we are just logging, we should be ok.

File: core/src/main/java/net/opentsdb/query/router/TimeRouterFactory.java (hunk 1 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/query/router/TimeRouterFactory.java]: should but I had a funky empty string popping up.

File: implementation/egads/src/main/java/net/opentsdb/query/anomaly/egads/olympicscoring/OlympicScoringNode.java (hunk 2 above)
Comment (manolama) [reply implementation/egads/src/main/java/net/opentsdb/query/anomaly/egads/olympicscoring/OlympicScoringNode.java]: yeah just logging

**Pull request merged**